py/scheduler.c

```diff
@@ -236,7 +236,13 @@ void mp_handle_pending(bool raise_exc) {
 
     // Handle any pending callbacks.
     #if MICROPY_ENABLE_SCHEDULER
-    if (MP_STATE_VM(sched_state) == MP_SCHED_PENDING) {
+    bool run_scheduler = (MP_STATE_VM(sched_state) == MP_SCHED_PENDING);
+    #if !MICROPY_PY_THREAD_GIL
```

Comment:
Is it worth making this `#if MICROPY_PY_THREAD && !MICROPY_PY_THREAD_GIL`, to be clear that it's only needed when threading is enabled?

I know it makes no difference, due to `mp_thread_is_main_thread()` being `false` when thread is disabled.  So could go either way here.

Reply:
I had the same thought, but erred on the other side - given we both thought it then I've made the change!